### PR TITLE
feat: support resource-specific Action endpoints (#295)

### PR DESCRIPTION
**Pull request by apricote**
Related to https://docs.hetzner.cloud/changelog#2023-06-29-resource-action-endpoints

The existing ActionClient has two purposes:

- Expose the GET /v1/actions & GET /v1/actions/:id endpoints
- Provider helper methods for waiting on running actions

The new resource-specific endpoints do not need the helper methods, so I have refactored the code for the endpoints into the ResourceActionClient. This client is exposed by all resources that have actions as client.Server.Action. In addition, I have replaced the implementation of the two endpoints in ActionClient by forwarding them to an instance of ResourceActionClient.

(cherry picked from commit ddc2ac45489c48a7563806c425222236ab1f8aa0 #295)

### CODE REVIEW

File: hcloud/action.go [hunk 1]
```diff
@@ -241,9 +198,9 @@ func (c *ActionClient) WatchOverallProgress(ctx context.Context, actions []*Acti
 				}
 			}
 
-			progress += (len(completedIDs) * 100)
+			progress += len(completedIDs) * 100
 			if progress != 0 && progress != previousProgress {
-				sendProgress(progressCh, int(progress/len(actions)))
+				sendProgress(progressCh, progress/len(actions))
```
Comment (jooola) [inline hcloud/action.go:203]: Why are those changes here ?

Comment (apricote) [reply hcloud/action.go:203]: Because they are part of the backported PR: https://github.com/hetznercloud/hcloud-go/pull/295/files#diff-8914d399a5e4a0ca8ac89a130609df3ba4006e707596cdbff04855cf929b2dcdR200-R202
